plugins/modules/jira.py

```diff
@@ -616,12 +624,23 @@ def operation_transition(self):
         turl = self.vars.restbase + '/issue/' + self.vars.issue + "/transitions"
         tmeta = self.get(turl)
 
-        target = self.vars.status
         tid = None
+        target = None
+
+        if self.vars.status is not None:
+            target = self.vars.status.strip()
+        elif self.vars.status_id is not None:
+            tid = self.vars.status_id.strip()
+
         for t in tmeta['transitions']:
-            if t['name'] == target:
-                tid = t['id']
-                break
+            if tid is None:
```

Comment:
I think
```suggestion
            if target is not None:
```
is easier to understand: if `target is not None`, then the user specified `status`. If not (in the else branch), the user specified `status_id` (stored in `tid`).

Reply:
Makes sense, thanks for pointing that out